crates/storage/src/connection/transaction.rs

```diff
@@ -435,27 +482,51 @@ pub(super) fn transaction_block_hash(
         .map_err(|e| e.into())
 }
 
-/// A copy of the gateway definitions which are currently used as the storage serde implementation. Having a copy here
-/// allows us to decouple this crate from the gateway types, while only exposing the common types via the storage API.
 pub(crate) mod dto {
     use fake::{Dummy, Fake, Faker};
     use pathfinder_common::*;
     use pathfinder_crypto::Felt;
-    use pathfinder_serde::{
-        CallParamAsDecimalStr, ConstructorParamAsDecimalStr, EthereumAddressAsHexStr,
-        L2ToL1MessagePayloadElemAsDecimalStr, ResourceAmountAsHexStr, ResourcePricePerUnitAsHexStr,
-        TipAsHexStr, TransactionSignatureElemAsDecimalStr,
-    };
     use serde::{Deserialize, Serialize};
-    use serde_with::serde_as;
+    use smallvec::SmallVec;
+
+    /// Minimally encoded Felt value.
+    #[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Default)]
+    #[serde(deny_unknown_fields)]
+    pub struct MinimalFelt(SmallVec<[u8; 32]>);
```

Comment:
Is there an advantage to using `SmallVec` over `[u8; 32]` here?

Reply:
I think it might be best to do so - `SmallVec` might change how it serde behaves without us realising it. And its non-obvious from the code why we're using it without knowing about its serde property.

Reply:
I don't think this actually saves space. You're using a backing store of 32 bytes anyway, and the full size of `MiminalFelt` is 48 bytes (on x86_64 at least).